samples/extensions/synchronization_2/synchronization_2.h

```diff
@@ -0,0 +1,120 @@
+/* Copyright (c) 2021, Sascha Willems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Doing compute and graphics queue synchronization using VK_KHR_synchronization2
+ * Synchronization examples using this extension can be found at https://github.com/KhronosGroup/Vulkan-Docs/wiki/Synchronization-Examples
+ */
+
+#pragma once
+
+#include "api_vulkan_sample.h"
+
+#if defined(__ANDROID__)
+// Lower particle count on Android for performance reasons
+#	define PARTICLES_PER_ATTRACTOR 3 * 1024
+#else
+#	define PARTICLES_PER_ATTRACTOR 4 * 1024
+#endif
+
+class Synchronization2 : public ApiVulkanSample
+{
+  public:
+	uint32_t num_particles;
+	uint32_t work_group_size  = 128;
+	uint32_t shared_data_size = 1024;
```

Comment:
I personally prefer to use initializer lists for easier reference (everything is in the cpp file for the construction; but I realize this is a personal preference; num_particles isn't inited).